scripts/asm-installer/asm_vm

```diff
@@ -1294,69 +1317,88 @@ EOF
   CREDENTIAL_IDENTITY_PROVIDER="${MEMBERSHIP_UID}@${WORKLOAD_IDENTITY_PROVIDER}@${WORKLOAD_NAMESPACE}"
 }
 
-# Generate GCE service proxy definition for the instance template
-gen_gce_service_proxy() {
-  local PROXY
+# A temporary solution to translate yaml to json in shell without relying on yq.
+yaml2json () {
+  # shellcheck disable=SC2016
+  ruby -r yaml -r json -e 'puts YAML.load($stdin.read).to_json'
```

Comment:
Just from personal experience, python2 is much more likely to be installed in systems already as a system dependency, so if there's a way to do it with code that works in python2 and python3 it would be more portable.

IMO It's probably not worth changing it unless the python implementation is just as straightforward as this ruby oneliner.

Reply:
Yep, just what Clay said.

There are multiple distribution of yq also. We could also write our own processor of yaml to json (probably based on sed), but I don't think it is worth the effort.

Ruby is the simplest solution I could think of and it is easy to install if not already.

Reply:
Then I am ok with ruby